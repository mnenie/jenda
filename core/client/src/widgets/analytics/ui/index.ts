import BoardsChart from "./BoardsChart.vue";
import OnlineChart from "./OnlineChart.vue";
import SharedSection from "./SharedSection.vue";
import TasksChart from "./TasksChart.vue";
import UsersChart from "./UsersChart.vue";

export { BoardsChart, OnlineChart, SharedSection, TasksChart, UsersChart };
