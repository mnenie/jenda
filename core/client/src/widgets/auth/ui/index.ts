import AuthContainer from './AuthContainer.vue'

export { AuthContainer }
