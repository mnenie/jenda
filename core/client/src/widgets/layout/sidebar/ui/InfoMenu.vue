<script setup lang="ts">
import { ref } from 'vue'
import { createReusableTemplate } from '@vueuse/core'
import { useExpanded } from '@/shared/composables'
import {
  UiBadge,
  UiButton,
  UiDropdownMenu,
  UiDropdownMenuContent,
  UiDropdownMenuGroup,
  UiDropdownMenuItem,
  UiDropdownMenuTrigger,
} from '@/shared/ui'
import { HotkeysDialog } from '@/widgets/dialogs'

defineProps<{
  isShowPlan?: boolean
}>()

const expanded = useExpanded()
const { isExpanded } = expanded.getExpanded()

const modalOpen = ref(false)

const [DifineTemplate, ReuseTemplate] = createReusableTemplate()
</script>

<template>
  <DifineTemplate v-slot="{ title, icon }">
    <UiDropdownMenuItem>
      <span class="text-sm" :class="icon" />
      <span class="2xl:text-13px text-sm font-medium">{{ title }}</span>
    </UiDropdownMenuItem>
  </DifineTemplate>
  <UiDropdownMenu direction="right">
    <UiDropdownMenuTrigger as-child>
      <UiButton
        v-tooltip.right="{
          content: $t(`sidebar.help.title'`),
          triggers: ['hover'],
          disabled: isExpanded,
        }"
        variant="ghost"
        class="w-full gap-2 shadow-none py-0 px-2 transition-all duration-200 ease justify-between mb-1"
        :class="[isExpanded ? 'justify-between' : 'justify-center p-0']"
      >
        <div class="flex items-center gap-2">
          <div i-hugeicons-mortarboard-02 class="!w-15px !h-15px 2xl:(!w-4 !h-4) text-neutral-800 dark:text-neutral-300" />
          <span v-show="isExpanded" class="text-13px 2xl:text-sm !fw500 text-neutral-900 dark:text-neutral-100">
            {{ $t('sidebar.help.title') }}
          </span>
        </div>
        <UiBadge v-if="isShowPlan" variant="soft" class="w-10 px-1">
          <div class="flex items-center gap-0.5">
            <div i-lucide-stars />
            <span class="!text-10px !fw500">1/3</span>
          </div>
        </UiBadge>
      </UiButton>
    </UiDropdownMenuTrigger>
    <UiDropdownMenuContent align="center" side="right" class="w-102%">
      <UiDropdownMenuGroup>
        <ReuseTemplate
          :title="$t('sidebar.help.items', 0)"
          icon="i-hugeicons-agreement-01"
        />
        <ReuseTemplate
          :title="$t('sidebar.help.items', 1)"
          icon="i-hugeicons-book-open-01"
          @click="modalOpen = true"
        />
        <ReuseTemplate
          :title="$t('sidebar.help.items', 2)"
          icon="i-hugeicons-link-forward"
        />
      </UiDropdownMenuGroup>
    </UiDropdownMenuContent>
  </UiDropdownMenu>
  <HotkeysDialog v-model:open="modalOpen" />
</template>
