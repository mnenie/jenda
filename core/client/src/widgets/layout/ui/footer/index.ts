import AppFooter from './AppFooter.vue'

export { AppFooter }
