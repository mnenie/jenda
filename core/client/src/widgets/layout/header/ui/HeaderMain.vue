<script setup lang="ts" generic="T extends Board">
import { computed, shallowReactive, toRef } from 'vue'
import { Icon } from '@iconify/vue'
import UserMenu from './UserMenu.vue'
import type { Board } from '@/entities/board'
import type { ProjectLink } from '@/shared/config/types-shared'
import { type User, UserAvatar } from '@/entities/user'
import { useLayoutPaths } from '@/shared/composables'
import { links } from '@/shared/constants'
import { HotkeysDialog, ShareDialog } from '@/widgets/dialogs'
import { ShimmerButton } from '@/shared/ui'

const props = defineProps<{
  projects: Board[]
}>()

const users = shallowReactive<User[]>([
  {
    _id: '0',
    email: 'alex@gmail.com',
    photoUrl: 'https://avatars.githubusercontent.com/u/121057011?v=4',
  },
  {
    _id: '1',
    email: 'airat@gmail.com',
    photoUrl: 'https://avatars.githubusercontent.com/u/95149637?s=100&v=4',
  },
  {
    _id: '2',
    email: 'slava@gmail.com',
    photoUrl: 'https://avatars.githubusercontent.com/u/83920644?v=4',
  },
  {
    _id: '3',
    email: 'egor@gmail.com',
    photoUrl: 'https://avatars.githubusercontent.com/u/84976745?v=4',
  },
  {
    _id: '4',
    email: 'egor@gmail.com',
    photoUrl: 'https://avatars.githubusercontent.com/u/84976745?v=4',
  },
  {
    _id: '5',
    email: 'egor@gmail.com',
    photoUrl: 'https://avatars.githubusercontent.com/u/84976745?v=4',
  },
])

const visibleUsers = computed(() => users.slice(0, 3))

const remaining = computed(() => Math.max(users.length - 3, 0))

const userPosition = computed(() => (index: number) => (visibleUsers.value.length - index) * 20)

const _projects = toRef(props, 'projects') as unknown as ProjectLink[]
const { active } = useLayoutPaths(links, _projects)
</script>

<template>
  <header
    class="sticky top-0 left-0 right-0 z-999 w-full flex items-center justify-between px-15px py-8px min-h-52px max-h-52px
    border-b border-b-solid border-layout"
  >
    <div class="inline-flex items-center gap-3">
      <Icon
        v-if="active.extendedAttrs.icon"
        :icon="active.extendedAttrs.icon"
        class="text-20px text-neutral-800 dark:text-neutral-200"
      />
      <div
        v-if="active.extendedAttrs.color"
        i-jenda-project
        :style="{ color: active.extendedAttrs.color, fontSize: '20px' }"
      />
      <p v-if="active" class="text-lg mt-1.5px !fw-600 dark:text-neutral-100">
        {{ active.name }}
      </p>
    </div>
    <div />
    <div class="flex items-center h-full gap-2.5">
      <div class="relative flex items-center">
        <UserAvatar
          v-if="remaining > 0"
          :style="{
            right: `0px`,
            zIndex: `${visibleUsers.length + 1}`,
          }"
          class="!h-28px !w-28px absolute flex items-center justify-center text-sm"
        >
          +{{ remaining }}
        </UserAvatar>
        <template v-for="user, idx in visibleUsers" :key="user._id">
          <UserAvatar
            :style="{
              right: `${userPosition(idx)}px`,
              zIndex: `${idx + 1}`,
            }"
            class="!h-28px !w-28px absolute"
          >
            <img :src="user.photoUrl" width="100%" />
          </UserAvatar>
        </template>
      </div>
      <ShareDialog>
        <ShimmerButton shimmer-size="2px">
          <div i-hugeicons-link-04 class="text-neutral-900 dark:text-neutral-100 w-16px h-16px 2xl:(w-4 h-4)" />
          <span
            class="whitespace-pre-wrap text-center text-neutral-800 dark:text-neutral-100 text-sm fw500"
          >
            {{ $t('header.share') }}
          </span>
        </ShimmerButton>
      </ShareDialog>
      <div
        v-tooltip="{ content: $t('header.navigator.messages'), trigger: ['hover'], distance: 7 }"
        i-hugeicons-message-multiple-01
        class="text-base text-neutral-700 cursor-pointer outline-none dark:text-neutral-300"
        style="margin-right: 2px"
      />
      <HotkeysDialog>
        <div
          v-tooltip="{ content: $t('header.navigator.question'), trigger: ['hover'], distance: 7 }"
          i-hugeicons-book-open-01
          class="text-base text-neutral-700 cursor-pointer outline-none dark:text-neutral-300"
          style="margin-right: -1px;"
        />
      </HotkeysDialog>
      <UserMenu />
    </div>
  </header>
</template>
