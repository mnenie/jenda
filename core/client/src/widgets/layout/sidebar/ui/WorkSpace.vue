<script setup lang="ts">
import { computed } from 'vue'
import { Icon } from '@iconify/vue'
import { useRoute } from 'vue-router/auto'
import type { WorkspaceLink } from '@/shared/config/types-shared'
import { useExpanded } from '@/shared/composables'
import { UiBadge, UiButton } from '@/shared/ui'

const props = defineProps<{
  links: WorkspaceLink[]
}>()

const route = useRoute()

function isCurrentPath(link: WorkspaceLink): boolean {
  if (route.name === 'boards') {
    return link.pathName === 'boards'
  }
  if (route.name === 'community') {
    return link.pathName === 'templates'
  }
  return route.path === `/${link.pathName}`
}

const pathName = computed(() => {
  return props.links.map(link => ({
    ...link,
    isActive: isCurrentPath(link),
  }))
})

const expanded = useExpanded()

const { isExpanded } = expanded.getExpanded()

const contentPosition = computed(() => {
  return isExpanded.value ? 'space-between' : 'center'
})
</script>

<template>
  <div class="flex flex-col space-y-1.5 justify-start mb-20px">
    <RouterLink
      v-for="link in pathName"
      :key="link.id"
      v-tooltip.right="{
        content: $t(`sidebar.${link.name}`),
        triggers: ['hover'],
        disabled: isExpanded,
      }"
      :to="`/${link.pathName}`"
      class="flex items-center gap-1.5 cursor-pointer text-neutral-900 justify-start w-full"
    >
      <UiButton
        :variant="link.isActive ? 'secondary' : 'ghost'"
        size="sm"
        class="w-full gap-2 shadow-none px-2 transition-all duration-200 ease"
        :class="link.isActive ? '!bg-neutral-200/40 dark:!bg-neutral-700/50' : ''"
        :style="{ padding: !isExpanded ? '0px' : '', justifyContent: contentPosition }"
      >
        <div class="flex items-center gap-2">
          <Icon :icon="link.icon" class="!w-15px !h-15px 2xl:(!w-4 !h-4) text-neutral-900 dark:text-neutral-200" />
          <span v-show="isExpanded" class="text-13px 2xl:text-sm !fw500 text-neutral-900 dark:text-neutral-100">{{ $t(`sidebar.${link.name}`) }}</span>
        </div>
        <UiBadge v-if="link.id === 1 && isExpanded" variant="solid" class="px-5px py-0 text-xs rounded-md">
          <span class="text-[10px] text-white">3</span>
        </UiBadge>
      </UiButton>
    </RouterLink>
  </div>
</template>
