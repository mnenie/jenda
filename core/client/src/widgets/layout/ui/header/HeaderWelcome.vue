<script setup lang="ts">
import type { Options } from '@/shared/ui/select/types'
import type { HeaderNavLink } from '../../model'
import { BurgerMenu } from '@/features/menu'
import { RouteNames } from '@/shared/config/consts'
import { useLanguage, useScroll } from '@/shared/lib/composables'
import { redirect } from '@/shared/lib/helpers'
import { UiButton, UiSelect } from '@/shared/ui'
import { useDark, useToggle, useWindowSize } from '@vueuse/core'
import { computed, ref, shallowReactive } from 'vue'
import { useI18n } from 'vue-i18n'
import { headerLinks } from '../../model'

const isDark = useDark()
const toggleDark = useToggle(isDark)

const { tm, t } = useI18n()

const iconUrl = computed(() => {
  return isDark.value ? '/icons/kanban-dark.png' : '/icons/kanban.png'
})

const languages = shallowReactive<Options[]>([
  { name: 'EN', value: 'en-US' },
  { name: 'RU', value: 'ru-RU' },
  { name: 'ZH', value: 'zh-CN' },
])
const language = ref('')

useLanguage(languages, language)

const links = computed(() => {
  const localeArr = tm('welcome.header.links') as HeaderNavLink[]
  return headerLinks.map((link, index) => ({
    ...link,
    name: localeArr[index],
  }))
})
const { scrollToEl } = useScroll(links.value.map(link => link.pagePrefix))
const { width } = useWindowSize()
</script>

<template>
  <header
    class="fixed top-0 left-0 right-0 z-50 flex flex-col self-center gap-10 max-w-1800px w-full px-4 py-3
    max-[1152px]:max-w-1064px max-[992px]:max-w-900px"
  >
    <slot name="offer" />
    <section class="flex items-center justify-between">
      <div class="flex items-center gap-2 select-none">
        <div
          v-if="width >= 1152"
          class="relative z-10 box bg-white border border-solid border-#e3e4e8 flex items-center py-0 pl-2.5 pr-3.5 h-40px rounded-lg
          dark:shadow-sm dark:bg-#313131 dark:border-neutral-700"
        >
          <div class="flex items-center gap-0.5">
            <div
              class="flex items-center gap-1 cursor-pointer mr-2"
              @click="redirect('https://github.com/mnenie/jenda')"
            >
              <img
                :src="iconUrl"
                class="w-22px h-22px"
              />
              <h3 class="text-xl">
                Jenda
              </h3>
            </div>
            <UiButton
              v-for="link in links"
              :key="link.id"
              variant="ghost"
              size="sm"
              class="text-neutral-500 font-medium hover:text-neutral-700 dark:text-neutral-400 dark:hover:!bg-neutral-700"
              @click="scrollToEl(link.pagePrefix)"
            >
              {{ link.name }}
            </UiButton>
          </div>
        </div>
        <div
          v-else
          class="flex items-center gap-1 cursor-pointer mr-2"
          @click="redirect('https://github.com/mnenie/jenda')"
        >
          <img
            :src="iconUrl"
            class="w-22px h-22px"
          />
          <h3 class="text-xl">
            Jenda
          </h3>
        </div>
        <div class="flex items-center gap-1">
          <UiSelect v-model="language" :options="languages" as="btn">
            <div i-hugeicons-globe-02 text-base />
          </UiSelect>
          <div class="w-1px h-3.5 bg-neutral-200 dark:bg-neutral-700" />
          <UiButton
            variant="ghost"
            size="sm"
            class="p-2 gap-1 text-neutral-400 hover:text-neutral-700 dark:!text-neutral-400 dark:hover:!text-neutral-200"
            @click="toggleDark()"
          >
            <div v-if="!isDark" i-hugeicons-moon-02 text-base />
            <div v-else i-lucide-sun text-base />
          </UiButton>
        </div>
      </div>
      <div v-if="width >= 1152" class="flex items-center gap-1.5 btns">
        <UiButton
          variant="ghost"
          style="font-weight: 500"
          @click="$router.push({ name: RouteNames.login })"
        >
          {{ t('welcome.header.login') }}
        </UiButton>
        <UiButton @click="$router.push({ name: RouteNames.registration })">
          {{ t('welcome.header.reg') }}
        </UiButton>
      </div>
      <BurgerMenu v-else />
    </section>
  </header>
</template>

<style scoped>
.box {
  box-shadow:
    0 12px 40px 0 rgba(0, 0, 0, 0.06),
    0 1px 0 0 rgba(0, 0, 0, 0.04);
}
</style>
