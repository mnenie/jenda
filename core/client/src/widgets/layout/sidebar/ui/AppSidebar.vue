<script setup lang="ts">
import { computed } from 'vue'
import { refDebounced } from '@vueuse/core'
import InfoMenu from './InfoMenu.vue'
import IntegrationItems from './IntegrationItems.vue'
import ProjectsList from './ProjectsList.vue'
import SearchBox from './SearchBox.vue'
import WorkSpace from './WorkSpace.vue'
import WorkspaceChooser from './WorkSpaceChooser.vue'
import type { Board } from '@/entities/board'
import { PlanCard } from '@/widgets/plan'
import { links } from '@/shared/constants'
import { useExpanded } from '@/shared/composables'

defineProps<{
  boards: Board[]
}>()

const expanded = useExpanded()

const { isExpanded } = expanded.getExpanded()

const debounced = refDebounced(isExpanded, 20)

const isShowPlan = computed((): boolean => {
  return isExpanded.value ? debounced.value : false
})

const paddingExpanded = computed(() => {
  return isExpanded.value ? '10px' : '0px 10px 10px 10px'
})
</script>

<template>
  <div class="relative w-full flex flex-col h-full bg-sidebar select-none border-r border-r-solid border-layout">
    <WorkspaceChooser />
    <div
      class="flex flex-col justify-between h-full mt-1.5"
      :style="{ padding: paddingExpanded }"
    >
      <div>
        <SearchBox :boards />
        <WorkSpace :links />
        <IntegrationItems />
        <ProjectsList :boards />
      </div>
      <div>
        <InfoMenu :is-show-plan />
        <PlanCard :is-show-plan />
      </div>
    </div>
  </div>
</template>
