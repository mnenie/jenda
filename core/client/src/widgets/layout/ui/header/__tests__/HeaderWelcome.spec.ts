import type { Options } from "@/shared/ui/select/types";
import type { VueWrapper } from "@vue/test-utils";
import i18n from "@/shared/lib/i18n";
import { UiButton, UiSelect } from "@/shared/ui";
import { mount } from "@vue/test-utils";
import { Moon, Sun } from "lucide-vue-next";
import { describe, expect, it, vi } from "vitest";
import { type ComponentPublicInstance, nextTick } from "vue";
import HeaderWelcome from "../HeaderWelcome.vue";
import "@/shared/lib/vitest-utils/cookiesI18n-mock";

type HeaderWelcomeInstance = ComponentPublicInstance<
  {},
  {},
  {
    isDark: boolean;
    width: number;
    languages: Options[];
    links: () => any[];
  }
>;

const mockRouter = {
  push: vi.fn(),
  beforeEach: vi.fn(),
};

describe("tests for HeaderWelcome.vue", () => {
  const wrapper = mount(HeaderWelcome, {
    global: {
      plugins: [i18n],

      mocks: {
        t: (key: string) => {
          const translations: Record<string, string> = {
            "welcome.header.login": "Log In",
            "welcome.header.reg": "Registration",
          };
          return translations[key];
        },
        tm: (key: string) => {
          const translationsArr: Record<string, string[]> = {
            "welcome.header.links": [],
          };
          return translationsArr[key];
        },
        $router: mockRouter,
      },
    },
  }) as VueWrapper<HeaderWelcomeInstance>;

  it("should be render correctly", async () => {
    wrapper.vm.width = 1600;
    await nextTick();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render subcomponents", async () => {
    wrapper.vm.width = 1600;
    await nextTick();
    expect(wrapper.findComponent(UiButton).exists()).toBe(true);
    expect(wrapper.findComponent(UiSelect).exists()).toBe(true);
  });

  it("should handle mode + icon changing", async () => {
    const darkModeButton = wrapper.find(".btn");
    wrapper.vm.isDark = false;
    wrapper.vm.width = 1600;
    await nextTick();

    expect(wrapper.findComponent(Moon).exists()).toBe(true);
    expect(wrapper.findComponent(Sun).exists()).toBe(false);

    darkModeButton.trigger("click");
    wrapper.vm.isDark = true;
    await nextTick();
    expect(wrapper.findComponent(Moon).exists()).toBe(false);
    expect(wrapper.findComponent(Sun).exists()).toBe(true);
  });

  it("should redirect correctly to login", async () => {
    const loginButton = wrapper.find(".btns").findAllComponents(UiButton).at(0);

    await loginButton.trigger("click");
    expect(mockRouter.push).toHaveBeenCalledWith({ name: "login" });
  });

  it("should redirect correctly to registration", async () => {
    const registrationButton = wrapper
      .find(".btns")
      .findAllComponents(UiButton)
      .at(1);

    await registrationButton.trigger("click");
    expect(mockRouter.push).toHaveBeenCalledWith({ name: "registration" });
  });
});
