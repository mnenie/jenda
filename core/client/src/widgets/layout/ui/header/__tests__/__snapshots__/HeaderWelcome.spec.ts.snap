// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tests for HeaderWelcome.vue > should be render correctly 1`] = `
"<header data-v-88b13e8b="" class="fixed top-0 left-0 right-0 z-50 flex flex-col self-center gap-10 max-w-1800px w-full px-4 py-3 max-[1152px]:max-w-1064px max-[992px]:max-w-900px">
  <section data-v-88b13e8b="" class="flex items-center justify-between">
    <div data-v-88b13e8b="" class="flex items-center gap-2 select-none">
      <div data-v-88b13e8b="" class="relative z-10 box bg-white border border-solid border-#e3e4e8 flex items-center py-0 pl-2.5 pr-3.5 h-40px rounded-lg dark:shadow-sm dark:bg-#313131 dark:border-neutral-700">
        <div data-v-88b13e8b="" class="flex items-center gap-0.5">
          <div data-v-88b13e8b="" class="flex items-center gap-1 cursor-pointer mr-2"><img data-v-88b13e8b="" src="/icons/kanban.png" class="w-22px h-22px">
            <h3 data-v-88b13e8b="" class="text-xl"> Jenda </h3>
          </div><button data-v-88b13e8b="" class="text-sm button ghost sm text-neutral-500 font-medium hover:text-neutral-700 dark:text-neutral-400 dark:hover:!bg-neutral-700">
            <!--v-if-->About
          </button><button data-v-88b13e8b="" class="text-sm button ghost sm text-neutral-500 font-medium hover:text-neutral-700 dark:text-neutral-400 dark:hover:!bg-neutral-700">
            <!--v-if-->Kanban
          </button><button data-v-88b13e8b="" class="text-sm button ghost sm text-neutral-500 font-medium hover:text-neutral-700 dark:text-neutral-400 dark:hover:!bg-neutral-700">
            <!--v-if-->Collaborative
          </button><button data-v-88b13e8b="" class="text-sm button ghost sm text-neutral-500 font-medium hover:text-neutral-700 dark:text-neutral-400 dark:hover:!bg-neutral-700">
            <!--v-if-->Activity
          </button><button data-v-88b13e8b="" class="text-sm button ghost sm text-neutral-500 font-medium hover:text-neutral-700 dark:text-neutral-400 dark:hover:!bg-neutral-700">
            <!--v-if-->Members
          </button><button data-v-88b13e8b="" class="text-sm button ghost sm text-neutral-500 font-medium hover:text-neutral-700 dark:text-neutral-400 dark:hover:!bg-neutral-700">
            <!--v-if-->Chats
          </button>
        </div>
      </div>
      <div data-v-88b13e8b="" class="flex items-center gap-1">
        <div data-v-88b13e8b="" class="btn_variant"><button class="text-sm button ghost sm btn">
            <!--v-if-->
            <div data-v-88b13e8b="" i-hugeicons-globe-02="" text-base=""></div> EN
          </button>
          <transition-stub name="dropdown" appear="false" persisted="false" css="true">
            <!--v-if-->
          </transition-stub>
        </div>
        <div data-v-88b13e8b="" class="w-1px h-3.5 bg-neutral-200 dark:bg-neutral-700"></div><button data-v-88b13e8b="" class="text-sm button ghost sm p-2 gap-1 text-neutral-400 hover:text-neutral-700 dark:hover:text-neutral-300">
          <!--v-if-->
          <div data-v-88b13e8b="" i-hugeicons-moon-02="" text-base=""></div>
        </button>
      </div>
    </div>
    <div data-v-88b13e8b="" class="flex items-center gap-1.5 btns"><button data-v-88b13e8b="" class="text-sm button ghost md" style="font-weight: 500;">
        <!--v-if-->Log In
      </button><button data-v-88b13e8b="" class="text-sm button default md">
        <!--v-if-->Registration
      </button></div>
  </section>
</header>"
`;
